### PR TITLE
Add schema version support

### PR DESCRIPTION
**Pull request by marcelveldt**
This belongs to https://github.com/zwave-js/zwave-js-server/pull/139

Instead of relying on server versions we switch to api schema to keep compatibility between clients and server versions.

### CODE REVIEW

File: zwave_js_server/client.py [hunk 1]
```diff
@@ -55,8 +54,16 @@ def connected(self) -> bool:
...
-    async def async_send_command(self, message: Dict[str, Any]) -> dict:
+    async def async_send_command(
+        self,
+        message: Dict[str, Any],
+        require_scheme: Optional[int] = None,
+    ) -> dict:
         """Send a command and get a response."""
+        if require_scheme is not None and require_scheme > self.scheme_version:
+            raise InvalidServerVersion(
+                "This command is not available, please update Z-Wave Server to a newer version."
```
Comment (MartinHjelmare) [inline zwave_js_server/client.py:65]: Include what version is required.

File: zwave_js_server/model/version.py [hunk 2]
```diff
@@ -18,4 +20,6 @@ def from_message(cls, msg: dict) -> "VersionInfo":
             driver_version=msg["driverVersion"],
             server_version=msg["serverVersion"],
             home_id=msg["homeId"],
+            min_scheme_version=msg.get("minSchemeVersion", 0),
```
Comment (MartinHjelmare) [inline zwave_js_server/model/version.py:23]: Let's add a comment about when the scheme versions were included in this message.

File: zwave_js_server/client.py (hunk 1 above)
Comment (marcelveldt) [reply zwave_js_server/client.py:65]: we require a server with supported scheme version, so not a specific version itself.
Should we log that ?

File: zwave_js_server/model/version.py (hunk 2 above)
Comment (marcelveldt) [reply zwave_js_server/model/version.py:23]: good idea

File: zwave_js_server/client.py (hunk 1 above)
Comment (MartinHjelmare) [reply zwave_js_server/client.py]: The minimum required version I mean.

Comment (marcelveldt) [reply zwave_js_server/client.py]: I've adjusted the message

File: zwave_js_server/const.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ class CommandClass(IntEnum):
     SCREEN_MD = 146
     SECURITY = 152
     SECURITY_2 = 159
-    SECURITY_SCHEME0_MARK = 61696
+    SECURITY_SCHEMA0_MARK = 61696
```
Comment (MartinHjelmare) [inline zwave_js_server/const.py:135]: This should probably not change.

Comment (marcelveldt) [reply zwave_js_server/const.py:135]: haha, find replace magic :-)

File: zwave_js_server/model/version.py [hunk 4]
```diff
@@ -18,4 +20,10 @@ def from_message(cls, msg: dict) -> "VersionInfo":
             driver_version=msg["driverVersion"],
             server_version=msg["serverVersion"],
             home_id=msg["homeId"],
+            # schema versions are sent in the response from schema version 1+
+            # this info not present means the server is at schema version 0
+            # at some point in time (when we stop supporting schema version 0),
+            # we could adjust this code and assume the keys are there.
+            min_schema_version=msg.get("minSchemaVersion", 0),
```
Comment (MartinHjelmare) [inline zwave_js_server/model/version.py:27]: I think we should add a test where we send data from a version 0 server, ie not including these new items. We want to check that we're allowing that here.

Comment (marcelveldt) [reply zwave_js_server/model/version.py:27]: Yes, I did not add any additional tests for added logic yet. I'll see what I can do later today.

**Review by balloob — APPROVED** [review]

**Comment by marcelveldt** [comment]
I could use some help writing the tests.

**Pull request merged**